lib/bot/cmds/thanks.js

```diff
@@ -21,37 +23,63 @@ const thanksCommands = {
 
     const mentions = input.message.model.mentions;
     // just 'thanks' in a message
-    if (mentions && mentions.length === 0) {
+    if (_.isEmpty(mentions)) {
       Utils.warn('thanks', 'without any mentions', input.message.model);
       return null;
     }
 
-    const fromUser = input.message.model.fromUser.username.toLowerCase();
+    const fromUser = input.message.model.fromUser.username;
+
     const options = {
       method: 'POST',
-      input: input,
-      bot: bot
+      input,
+      bot
     };
 
-    const namesList = mentions.reduce((userList, mention) => {
-      const toUser = mention.screenName.toLowerCase();
-      if (toUser !== fromUser && userList.indexOf(toUser) === -1) {
-        const apiPath = '/api/users/give-brownie-points?receiver=' + toUser +
-                        '&giver=' + fromUser;
-        HttpWrap.callApi(apiPath, options, thanksCommands.showInfoCallback);
-        userList.push(toUser);
-      }
-      return userList;
-    }, []);
-
-    if (namesList.length > 0) {
-      const toUserMessage = namesList.join(' and @');
-      return '> ' + fromUser + ' sends brownie points to @' + toUserMessage +
-        ' :sparkles: :thumbsup: :sparkles: ';
-    } else {
-      return '> sorry ' + fromUser + ', you can\'t send brownie points to ' +
-        'yourself! :sparkles: :sparkles: ';
-    }
+    const thankList = _.chain(mentions)
+      .uniq('screenName')
+      .filter((user) =>
+        user.screenName.toLowerCase() !== fromUser.toLowerCase()
+        && !!user.userId
+      )
+      .map((user) => user.screenName)
+      .value();
+
+     thankList.forEach((toUser) => {
+       const apiPath = `/api/users/give-brownie-points?receiver=${toUser.toLowerCase()}&giver=${fromUser.toLowerCase()}`;
+       HttpWrap.callApi(apiPath, options, thanksCommands.showInfoCallback);
+     });
+
+     let message = '';
+
+     const THANKLIST_SIZE = thankList.length;
+     if (THANKLIST_SIZE > 0) {
+       const $symboledThankedList = _.chain(thankList)
+          .map((user) => `@${user}`);
+
+       const $lastUser = $symboledThankedList.last();
+       const $initialUsers = $symboledThankedList.take(THANKLIST_SIZE - 1);
+
+       let thankedUsersMsg = $initialUsers.value().join(', ');
+       // if more than one user is thanked
+       // then append an "and"
+       if (thankedUsersMsg !== '') {
```

Comment:
You may reuse the `THANKLIST_SIZE` constant here which would more appropriate and better i.e. 
```js
THANKLIST_SIZE > 1
```